### PR TITLE
TRIDENT-3476: end-to-end consume+demux works

### PR DESCRIPTION
**Pull request by jasklmclane**
From HTTPie:
`http POST localhost:8080/bro/flow < ./bluecoat_tagged.log`

I get Syslog-NG output:

{
  "source": {
    "username": "sarah",
    "userAgent": "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.107 Safari/537.36",
    "ip": "192.168.1.64",
    "group": "-"
  },
  "http": {
    "status": "200",
    "scheme": "http",
    "result": "PROXIED",
    "referer": "http://tappedout.net/mtg-combos/bident-of-thassa%2Bthopter-spy-network/",
    "query": "-",
    "path": "/gpt/pubads_impl_68r2.js",
    "method": "GET",
    "extension": "js",
    "contentType": "text/javascript",
    "category": "Web Ads/Analytics",
    "cacheAction": "TCP_MISS",
    "bytes": "48551"
  },
  "event": {
    "virusId": "521",
    "time": "20:53:02",
    "exceptionId": "-",
    "duration": "1552",
    "date": "2015-07-28"
  },
  "destination": {
    "port": "80",
    "ip": "192.168.1.190",
    "host": "partner.googleadservices.com"
  },
  "TAGS": ".source.s_bluecoat",
  "SOURCEIP": "127.0.0.1",
  "SOURCE": "s_bluecoat",
  "PRIORITY": "notice",
  "MESSAGE": "2015-07-28 20:53:02 1552 192.168.1.64 sarah - - PROXIED \"Web Ads/Analytics\" http://tappedout.net/mtg-combos/bident-of-thassa%2Bthopter-spy-network/  200 TCP_MISS GET text/javascript http partner.googleadservices.com 80 /gpt/pubads_impl_68r2.js - js \"Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.107 Safari/537.36\" 192.168.1.190 48551 521 - \"none\" \"none\"",
  "HOST_FROM": "Lucass-MacBook-Pro",
  "HOST": "Lucass-MacBook-Pro",
  "FILE_NAME": "/tmp/syslog-ng/bluecoat",
  "FACILITY": "user",
  "DATE": "Jun 18 15:32:04"
}

### CODE REVIEW

File: parse/src/main/scala/io/jask/svc/northstar/Demux.scala [hunk 1]
```diff
@@ -1,25 +1,80 @@
...
+  def buildOutputsMap(): Map[String, FileOutputStream] = {
+    var outputsMap:Map[String, FileOutputStream] = Map()
+    val d = new File(config.getString("northstar.parser.folder"))
+    if (d.exists && d.isDirectory) {
+      for(file <- d.listFiles.toList) {
+        try {
+         outputsMap += (file.getName -> new FileOutputStream(file))
+         log.info("Adding demux output: " + file.getName)
+        } catch {
+          case e: Throwable => {
```
Comment (derrley) [inline parse/src/main/scala/io/jask/svc/northstar/Demux.scala]: hm. Catch all? What is the actual error that tends to happen here.

File: parse/src/main/scala/io/jask/svc/northstar/Demux.scala [hunk 2]
```diff
@@ -1,25 +1,80 @@
...
+            log.error("Skipping output: " + file.getName + " " + e.getMessage)
+          }
+        }
+      }
+      outputsMap
```
Comment (derrley) [inline parse/src/main/scala/io/jask/svc/northstar/Demux.scala]: Let me pair up with you and teach you how to write this without mutation :)

**Comment by jasklmclane** [comment]
Cool.... yeah we'll chat more Monday, but I added some more code to this PR as well after more rigorous testing.

File: parse/src/main/scala/io/jask/svc/northstar/Demux.scala (hunk 1 above)
Comment (jasklmclane) [reply parse/src/main/scala/io/jask/svc/northstar/Demux.scala]: So far I've only seen IOException: Permission Denied (as the e.message which get's log.error()'d)

**Pull request merged**